### PR TITLE
wheels: drop py36 and add py310 support

### PR DESCRIPTION
**Pull request by alexsavulescu**
TODO:
- [x] Relaunch CI once https://github.com/BlueBrain/CoreNeuron/pull/840 is in and neuron submodule updated

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -264,7 +264,7 @@ stages:
           -w /root/nmodl \
           -v $PWD:/root/nmodl \
           -e NMODL_NIGHTLY_TAG=$TAG \
-          'bluebrain/nmodl:wheel' \
+          'bluebrain/nmodl:wheel-test' \
```
Comment (alexsavulescu) [inline azure-pipelines.yml:267]: TODO: revert this and upload docker image before merge

**Review by ohm314 — APPROVED** [review]

File: azure-pipelines.yml [hunk 2]
```diff
@@ -5,7 +5,7 @@ variables:
   # defined in UI
   #- ReleaseWheelBuild: False
   #- UploadWheel: False
-  buildWheel: ${{ or(in(variables['Build.Reason'], 'Schedule'), in(variables['Build.Reason'], 'Manual')) }}
+  buildWheel: True #TODO: revertthis ${{ or(in(variables['Build.Reason'], 'Schedule'), in(variables['Build.Reason'], 'Manual')) }}
```
Comment (alexsavulescu) [inline azure-pipelines.yml:8]: .

Comment (alexsavulescu) [reply azure-pipelines.yml:8]: Only 3.9 gets built for PRs..

Comment (olupton) [reply azure-pipelines.yml:8]: Sorry, right. I got confused by the micro-diff.

Comment (alexsavulescu) [reply azure-pipelines.yml:8]: _keeping this unresolved to avoid merging by mistake_

File: azure-pipelines.yml [hunk 3]
```diff
@@ -288,22 +292,28 @@ stages:
...
+            python.installer.name: 'macosx10.9.pkg'
+          Python310:
+            python.version: '3.10'
+            python.org.version: '3.10.5'
+            python.installer.name: 'macos11.pkg'
+
         ${{ if eq(variables.buildWheel, False) }}:
           Python39:
             python.version: '3.9'
             python.org.version: '3.9.0'
```
Comment (olupton) [inline azure-pipelines.yml:315]:
```suggestion
            python.org.version: '3.9.13'
```

File: azure-pipelines.yml [hunk 4]
```diff
@@ -162,7 +166,7 @@ stages:
           make VERBOSE=1
           exit 1
         fi
-        ctest --output-on-failure
+        ctest -VV #--output-on-failure
```
Comment (alexsavulescu) [inline azure-pipelines.yml:169]: TODO - revert this

File: azure-pipelines.yml (hunk 1 above)
Comment (olupton) [reply azure-pipelines.yml:272]: In the end, what changes are needed in the image?

**Review by olupton — APPROVED** [review]
Changes LGTM apart from the TODOs you flagged. Note this is using the `aaa` branch of NEURON, which reverts the use of `%a`...is that also needed?

**Comment by alexsavulescu** [comment]
> Changes LGTM apart from the TODOs you flagged. Note this is using the `aaa` branch of NEURON, which reverts the use of `%a`...is that also needed?

Not needed. Thanks.

**Comment by alexsavulescu** [comment]
> In the end, what changes are needed in the image? 

We just needed a newer build (also to get Python3.10), didn't change anything in the Dockerfile

**Pull request merged**